service-scripts.xml

```diff
@@ -1,160 +1,171 @@
-            <es:set-value target="SSNMask" value="//User/SSNMask"/>
-          </es:when>
-        </es:choose>
-      </es:when>
     </es:choose>
   </es:transform>
   <es:transform name="UserInBand" target="soap:Body/{$query}/{$request}/Row/User">
+    <!--
+    Values set in this transform are set for a high percentage of scripted requests that
+    embed a shared User structure. No value referenced here is either method or service specific.
+
+    This does not apply to OscarSearch.
+    -->
     <es:choose>
-      <es:otherwise>
-        <es:choose>
-          <es:otherwise>
-            <es:set-value target="GLBPurpose" value="&apos;0&apos;"/>
-          </es:otherwise>
-          <es:when test="string-length(normalize-space(//GLBPurpose))=0"/>
-        </es:choose>
-      </es:otherwise>
       <es:when test="contains($glbAllowed, concat(&apos; &apos;, string(number(//GLBPurpose)), &apos; &apos;))">
         <es:set-value target="GLBPurpose" value="string(number(//GLBPurpose))"/>
       </es:when>
+      <es:when test="not(string-length(normalize-space(//GLBPurpose))=0)">
+        <es:set-value target="GLBPurpose" value="&apos;0&apos;"/>
+      </es:wehn>
     </es:choose>
     <es:choose>
-      <es:otherwise>
-        <es:choose>
-          <es:otherwise>
-            <es:set-value target="DLPurpose" value="&apos;0&apos;"/>
-          </es:otherwise>
-          <es:when test="string-length(normalize-space(//DLPurpose))=0"/>
-        </es:choose>
-      </es:otherwise>
       <es:when test="contains($dppaAllowed, concat(&apos; &apos;, string(number(//DLPurpose)), &apos; &apos;))">
         <es:set-value target="DLPurpose" value="string(number(//DLPurpose))"/>
       </es:when>
+      <es:when test="not(string-length(normalize-space(//DLPurpose))=0)">
+        <es:set-value target="DLPurpose" value="&apos;0&apos;"/>
+      </es:when>
     </es:choose>
   </es:transform>
   <es:transform name="BertErniePlatformOutOfBand" target="soap:Body/{$query}">
+    <!--
+    The values set in this transformation are product specific values that apply to all
+    service methods with the exception of OscarSearch.
+    -->
+    <es:set-value target="LogInfo/ProductId" value="&apos;7&apos;"/>
+    <es:set-value target="LogInfo/SubProductId" value="&apos;70014&apos;"/>
+  </es:transform>
+  <es:transform name="BertErniePlatformInBand" target="soap:Body/{$query}/{$request}/Row">
+    <!--
+    The values set in this transformation are product specific values that apply to both
+    BertErnieSearch and BertErnieReport.
+    -->
     <es:choose>
-      <es:when test="true()">
-        <es:append-to-value target="LogInfo/ProductId" value="&apos;7&apos;"/>
-        <es:append-to-value target="LogInfo/SubProductId" value="&apos;70014&apos;"/>
+      <es:when test="$internal_user=1">
+        <es:if test="string-length(ProductCode)=0">
+          <es:set-value name="prod_i" optional="false" target="BertErnieUser/ProductCode" value="&apos;10&apos;"/>
```

Comment:
I don't recall, what does the @optional do?

Reply:
It controls whether evaluation failures terminate the transaction (false) or are ignored (true). Once refactored, it will only be supported for legacy markup. Failures will be failures.